src/biogeophys/OzoneMod.F90

```diff
@@ -547,6 +692,118 @@ subroutine CalcOzoneStress(this, bounds, &
+         o3coefgsha  => this%o3coefgsha_patch                 , & ! Output: [real(r8) (:)] ozone coef
+         o3coefgsun  => this%o3coefgsun_patch                   & ! Output: [real(r8) (:)] ozone coef
+         )
+
+      do fp = 1, num_exposedvegp
+         p = filter_exposedvegp(fp)
+
+         ! Ozone stress for shaded leaves
+         call CalcOzoneStressLi2024OnePoint( &
+              pft_type=patch%itype(p), o3uptake=o3uptakesha(p), &
+              o3coefv=o3coefvsha(p), o3coefg=o3coefgsha(p))
+
+         ! Ozone stress for sunlit leaves
+         call CalcOzoneStressLi2024OnePoint( &
+              pft_type=patch%itype(p), o3uptake=o3uptakesun(p), &
+              o3coefv=o3coefvsun(p), o3coefg=o3coefgsun(p))
+      end do
+
+      do fp = 1, num_noexposedvegp
+         p = filter_noexposedvegp(fp)
+
+         ! See notes above in InitHistory about why these need to be set to 1 over
+         ! non-exposed veg points each time step.
+         o3coefvsha(p) = 1._r8
+         o3coefgsha(p) = 1._r8
+         o3coefvsun(p) = 1._r8
+         o3coefgsun(p) = 1._r8
+      end do
+
+    end associate
+
+  end subroutine CalcOzoneStressLi2024
+  
+  !----------------------------------------------------------------------
+   subroutine CalcOzoneStressLi2024OnePoint( &
+       pft_type, o3uptake, &
+       o3coefv, o3coefg)
+    !
+    ! !DESCRIPTION:
+    ! Calculates ozone stress for a single point, for just sunlit or shaded leaves
+    !
+    ! This subroutine uses the Li2024 formulation for ozone stress
+    !
+    ! !ARGUMENTS:
+    integer  , intent(in)    :: pft_type   ! vegetation type, for indexing into pftvarcon arrays
+    real(r8) , intent(in)    :: o3uptake   ! ozone entering the leaf
+    real(r8) , intent(out)   :: o3coefv    ! ozone coefficient for photosynthesis (0 - 1)
+    real(r8) , intent(out)   :: o3coefg    ! ozone coefficient for conductance (0 - 1)
+    !
+
+    character(len=*), parameter :: subname = 'CalcOzoneStressLi2024OnePoint'
+    !-----------------------------------------------------------------------
+
+    if (o3uptake == 0._r8) then
+       ! No o3 damage if no o3 uptake
+       o3coefv = 1._r8
+       o3coefg = 1._r8
+    else
+       ! Determine parameter values for this pft
+      if(pft_type >= 1 .and. pft_type <= 3)then  !Needleleaf tree
+        o3coefv = max(0._r8, min(1._r8, 1.005_r8 - 0.0064_r8 * o3uptake))
+        o3coefg = max(0._r8, min(1._r8, 0.965_r8 * o3uptake ** (-0.041)))
+      end if
+      if(pft_type >= 4 .and. pft_type <= 8)then  !Broadleaf tree
+        o3coefv = max(0._r8, min(1._r8, 0.943_r8 * exp(-0.0085*o3uptake)))
+        o3coefg = max(0._r8, min(1._r8, 0.943_r8 * exp(-0.0058*o3uptake)))
+      end if
+      if(pft_type >= 9 .and. pft_type <= 11)then !Shrub
+        o3coefv = max(0._r8, min(1._r8, 1.000_r8-0.074_r8 * log(o3uptake)))
+        o3coefg = max(0._r8, min(1._r8, 0.991_r8-0.060_r8 * log(o3uptake)))
```

Comment:
These right-hand-sides also do not match what I see in Li et al. (in prep), which says
```
1.142 POD5**(-0.140)
1.075 POD5**(-0.104)
```

Reply:
see above reply